languages/js/src/types.ts

```diff
@@ -600,10 +600,7 @@ export function isIterableIterator(x: unknown): x is IterableIterator<unknown> {
  * @internal
  */
 export function isIterable(x: unknown): x is Iterable<unknown> {
-  return (
-    Symbol.iterator in Object(x) &&
-    typeof Object(x)[Symbol.iterator] === 'function'
-  );
+  return isObj(x) && Symbol.iterator in x;
```

Comment:
same as `asyncIterator` below

Reply:
Responded in https://github.com/osohq/oso/pull/1227#discussion_r715900928; let's discuss there.